app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/Events.kt

```diff
@@ -187,81 +185,110 @@ fun Events(nav: NavigationActions, userViewModel: UserViewModel, eventViewModel:
               if (!eventsLoaded.value) {
                 LoadingText()
               } else {
+                val joinedEvents =
+                    eventList.filter { e -> user.value!!.joinedEvents.contains(e.eventID) }
+                val favouriteEvents =
+                    eventList.filter { e -> user.value!!.myFavorites.contains(e.eventID) }
+                val myEvents = eventList.filter { e -> e.creator == user.value!!.uid }
 
-                // Display events based on the selected filter
-                Column(
-                    modifier = Modifier.verticalScroll(rememberScrollState()).fillMaxSize(),
-                    horizontalAlignment = Alignment.CenterHorizontally) {
-
-                      // Display joined events if 'All' or 'Joined' is selected
-                      if (selectedFilter == ALL || selectedFilter == JOINED) {
-                        Spacer(modifier = Modifier.height(screenHeight / 40))
-                        Text(
-                            text = JOINED.formattedName,
-                            style = MaterialTheme.typography.titleLarge,
-                            color = MaterialTheme.colorScheme.onBackground,
-                            modifier =
-                                Modifier.padding(horizontal = screenWidth / 15)
-                                    .testTag("JoinedTitle"))
+                if (joinedEvents.isEmpty() && favouriteEvents.isEmpty() && myEvents.isEmpty()) {
+                  Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
+                    val annotatedText = buildAnnotatedString {
+                      withStyle(style = SpanStyle(color = MaterialTheme.colorScheme.onBackground)) {
+                        append("You don't have any events yet, explore ")
```

Comment:
I think that these "magic strings" (lines 198-210) are justified, since it's a message that's used once and is separated into 3 substrings of different text styles that get appended together.